codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -24,184 +24,37 @@
-            return value.charAt(0);
-        }
-
-        @Override
-        public CharSequence convertShort(short value) {
-            return String.valueOf(value);
-        }
-
-        @Override
-        public short convertToShort(CharSequence value) {
-            return Short.valueOf(value.toString());
-        }
-
-        @Override
-        public int convertToInt(CharSequence value) {
-            return Integer.parseInt(value.toString());
-        }
-
-        @Override
-        public long convertToLong(CharSequence value) {
-            return Long.parseLong(value.toString());
-        }
-
-        @Override
-        public AsciiString convertTimeMillis(long value) {
-            return new AsciiString(String.valueOf(value));
-        }
-
-        @Override
-        public long convertToTimeMillis(CharSequence value) {
-            try {
-                return HeaderDateFormat.get().parse(value.toString());
-            } catch (ParseException e) {
-                PlatformDependent.throwException(e);
-            }
-            return 0;
-        }
-
-        @Override
-        public float convertToFloat(CharSequence value) {
-            return Float.valueOf(value.toString());
-        }
 
-        @Override
-        public double convertToDouble(CharSequence value) {
-            return Double.valueOf(value.toString());
-        }
-    }
-
-    private static final ValueConverter<CharSequence> CHARSEQUENCE_FROM_OBJECT_CONVERTER =
-            new DefaultTextValueTypeConverter();
-    private static final TypeConverter<CharSequence, String> CHARSEQUENCE_TO_STRING_CONVERTER =
-            new TypeConverter<CharSequence, String>() {
-        @Override
-        public String toConvertedType(CharSequence value) {
-            return value.toString();
-        }
-
-        @Override
-        public CharSequence toUnconvertedType(String value) {
-            return value;
-        }
-    };
-
-    private static final NameConverter<CharSequence> CHARSEQUENCE_IDENTITY_CONVERTER =
-            new IdentityNameConverter<CharSequence>();
-    /**
-     * An estimate of the size of a header value.
-     */
-    private static final int DEFAULT_VALUE_SIZE = 10;
+    public static final NameValidator<CharSequence> NO_NAME_VALIDATOR = NoNameValidator.instance();
 
     private final ValuesComposer valuesComposer;
 
     public DefaultTextHeaders() {
-        this(true);
-    }
-
-    public DefaultTextHeaders(boolean ignoreCase) {
-        this(ignoreCase, CHARSEQUENCE_FROM_OBJECT_CONVERTER, CHARSEQUENCE_IDENTITY_CONVERTER, false);
+        this(false);
     }
 
-    public DefaultTextHeaders(boolean ignoreCase, boolean singleHeaderFields) {
-        this(ignoreCase, CHARSEQUENCE_FROM_OBJECT_CONVERTER, CHARSEQUENCE_IDENTITY_CONVERTER, singleHeaderFields);
+    public DefaultTextHeaders(boolean singleHeaderFields) {
+        this(new TreeMap<CharSequence, Object>(CHARSEQUENCE_CASE_INSENSITIVE_ORDER), NO_NAME_VALIDATOR,
```

Comment:
Good question.  I think there is an opportunity for optimization here. However if we are getting performance benefits from simplifying perhaps this could be done in a followup PR?  @buchgr - If you don't want to have an optimized comparator in this PR then maybe just create a follow up issue to investigate?